app/views/exams/ProgressPage.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <div class="progress-page center-div">
+    <div class="header">
+      <div class="code-language">
+        {{ userExam.codeLanguage }}
+      </div>
+      <div class="timer">
+        Time Left: {{ timeLeft }}
+      </div>
+    </div>
+    <div class="levels">
+      <exam-level
+        v-for="(level, index) in problems"
+        :key="level._id"
+        :level="level"
+        :index="index + 1"
+      />
+    </div>
+
+    <div class="submit center-div">
+      <input
+        type="button"
+        value="Mark as Complete"
+        @click="submit"
+      >
+      <div class="tip">
+        Are you sure you want to mark as complete? You will not be able to play the levels again.
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapGetters, mapActions } from 'vuex'
+import ExamLevel from './components/ExamLevel'
+export default {
+  components: {
+    ExamLevel,
+  },
+  data () {
+    return {
+      timeLeft: '00:00',
+      counterInterval: null,
+    }
+  },
+  computed: {
+    ...mapGetters('exams', [
+      'getExamById',
+      'userExam',
+    ]),
+    exam () {
+      return this.getExamById(this.examId)
+    },
+    problems () {
+      const problems = this.exam.problems
```

Comment:
code should be able to work when this.exam isn't fetched too instead of throwing exception